File: src/dataset/cpp/difference-of-squares_exercism-nayefc/claude/ClaudeSonnet4_difference_of_squares_v4.cpp
#include "difference_of_squares.h"

int squares::square_of_sums(int n) {
    int sum = n * (n + 1) / 2;
    return sum * sum;
}

int squares::sum_of_squares(int n) {
    return n * (n + 1) * (2 * n + 1) / 6;
}

int squares::difference(int n) {
    int sum = n * (n + 1) / 2;
    int square_of_sums = sum * sum;
    int sum_of_squares = n * (n + 1) * (2 * n + 1) / 6;
    return square_of_sums - sum_of_squares;
}

File: src/dataset/javascript/difference-of-squares_exercism-ThomasZumsteg/claude/ClaudeSonnet4_difference-of-squares_v4.js
var Squares = function(num) {
	var numPlusOne = num + 1;
	var numSquared = num * num;
	var numPlusOneSquared = numPlusOne * numPlusOne;
	
	this.squareOfSums = (numSquared * numPlusOneSquared) >> 2;
	this.sumOfSquares = (num * numPlusOne * ((num << 1) + 1)) / 6;
	this.difference = this.squareOfSums - this.sumOfSquares;
}

module.exports = Squares;